autocomplete.js

```diff
@@ -1,70 +1,48 @@
-const GoogleCloudStorage = require('./google-cloud-services')
-const parsers = require('./parsers')
-
-const MAX_RESULTS = 25;
-
-// auto complete helper methods
-
-function mapAutoParams(autoParams){
-  const params = {};
-  autoParams.forEach(param => {
-    params[param.name] = parsers.autocomplete(param.value);
-  });
-  return params;
-}
-
-/***
- * @returns {[{id, value}]} filtered result items
- ***/
-function handleResult(result, query, keyField, valField){
-  if (!result || result.length == 0) return [];
-  if (!keyField) {
-    keyField = "id";
-    valField = "name";
-  }
-  const items = result.map(item => getAutoResult(item[keyField], item[valField]));
-  return filterItems(items, query);
-}
-
-/***
- * @returns {{id, value}} formatted autocomplete item
- ***/
-function getAutoResult(id, value) {
-  return {
-    id: id || value,
-    value: value || id
+const GoogleCloudStorageClient = require("@google-cloud/storage");
+const _ = require("lodash");
+
+function createAutocompleteFunction(fetchItems, [valuePath, idPath = ""]) {
```

Comment:
Why array, and not an object, which seems more natural for a finite, defined set of properties of common nature.